### DISCUSSION TITLE
Can Bagetter be installed on Linux server?

### DISCUSSION THREAD

**Discussion by RobSmyth** [bagetter/BaGetter#167] [General]
Hi,

The title says it, can I install Bagetter on a linux server?

Thx

**Comment by Regenhardt** [comment]
Hi 

You absolutely can! The easiest way would be using the docker image and running that. There are both x64 and ARM images on docker hub. Configure them to your liking by setting environment variables. 

You could also run it using the dotnet runtime, creating a service to make sure it's started on startup.  
I'm not familiar with built-in Linux web servers unfortunately, so to do it without docker I guess it depends on your distro?  
Basically Bagetter comes as a dotnet web service (the executable is called Bagetter), you can use a general guide to host it. 

Let me know if you have any more questions, I'll try to answer them if I can. I definitely recommend using a container runtime like docker or podman though.

**Reply by RobSmyth** [reply]
Thank you ... I will give it a go.

**Comment by RobSmyth** [comment]
Yes, it worked well. I installed it on a linux VM. I probably did not it the best way but here is how I (who knows nothing about docker and very little about linux) got it to work ...

After testing the docker container with the given `docker run` command I assigned it a container name `bagetter` (I used a different data folder `/var/bagetter/data`):

```
docker run --name bagetter -p 5000:8080 --env-file /var/bagetter/data/bagetter.env -v "/var/bagetter/data:/data" bagetter/bagetter:latest
```

Then, so it boots on start, I added a systemd file `/etc/systemd/system/bagetter.service`:

```
[Unit]
Description=BaGetter NuGet server
Requires=docker.service
After=docker.service

[Service]
Restart=always
ExecStart=/usr/bin/docker start -a bagetter
ExecStop=/usr/bin/docker stop -t 2 bagetter

[Install]
WantedBy=multi-user.target
```

Not sure if I needed to but I then did:

```
sudo systemctl enable bagetter
```

It now starts automatically on reboot and can be started manually by:

```
sudo systemctl start bagetter
```

This was done on a Proxmox LXC created by the script from: https://tteck.github.io/Proxmox/#docker-lxc.

**Reply by Regenhardt** [reply]
Great, thanks for the update, this might help other people too.
